courses/fundamentals_of_ada/labs/answers/spec_892_ada_strings.txt

```diff
@@ -0,0 +1,100 @@
+--declarations
+with Ada.Text_IO; use Ada.Text_IO;
+with Ada.Strings.Fixed;
+with Ada.Strings.Unbounded;
+procedure Main is
+
+   -- hard-coded filename
+   Filename : constant String := "main.adb";
+
+   File : File_Type;
+   Line : String (1 .. 100);
+   Last : Natural;
+
+   Objects      : array (1 .. 100) of Ada.Strings.Unbounded.Unbounded_String;
+   Object_Count : Natural := 0;
+
+   Comments   : Natural := 0;
+   Semicolons : Natural := 0;
+
+   Colon : Natural;
+
+--declarations
+--extra_credit
+   function Hide_Strings
+     (Str : String)
+      return String is
+      First : Natural;
+      Last  : Natural;
+   begin
+      First := Ada.Strings.Fixed.Index (Str, """");
+      if First in Str'range then
+         Last := Ada.Strings.Fixed.Index
+             (Source  => Str (First + 1 .. Str'last),
+              Pattern => """");
+         if Last in Str'range then
+            return Ada.Strings.Fixed.Replace_Slice (Str,
```

Comment:
Multiline args indent style is incorrect, result is weird, should go to the line before the opening parenthesis

Reply:
Not sure what you mean. Line 35 should be lined up with Line 32 (and it is)

Reply:
line 36 and 37, the columns are crushed to the right

Reply:
This is what "gnatpp" does in a return (I guess) - I didn't make it look like that :)
That being said, I will clean it u